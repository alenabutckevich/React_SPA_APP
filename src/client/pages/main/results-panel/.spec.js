import React from 'react';
import toJson from 'enzyme-to-json';
import ResultsPanel from './results-panel';

const setup = propOverrides => {
    const { movies } = propOverrides;
    const props = Object.assign({
        movies: movies
    }, propOverrides);

    const wrapper = shallow(<ResultsPanel {...props}/>);

    return {
        wrapper,
        props
    }
}

describe('<ResultsPanel />', () => {

    it('render results panel', () => {
        const movies = [{
            id:1, title:'title1', poster_path:'path1', genre_ids:[1], release_date:'2012-12-12'
        }, {
            id:2, title:'title2', poster_path:'path2', genre_ids:[2], release_date:'2012-12-12'
        }];
        const { wrapper } = setup({movies: movies});
        expect(wrapper.find('.results_panel__movies_count').text()).toEqual('2 movies found');

        expect(toJson(wrapper)).toMatchSnapshot();
    })

    it('no render results panel', () => {
        const { wrapper } = setup([]);
        expect(wrapper.find('.results_panel__movies_count')).toHaveLength(0);

        expect(toJson(wrapper)).toMatchSnapshot();
    })
})