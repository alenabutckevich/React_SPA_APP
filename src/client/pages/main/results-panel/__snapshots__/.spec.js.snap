// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ResultsPanel /> no render results panel 1`] = `
<div
  className="results_panel"
/>
`;

exports[`<ResultsPanel /> render results panel 1`] = `
<div
  className="results_panel"
>
  <div>
    <span
      className="results_panel__movies_count"
    >
      2
       movies found
    </span>
    <ResultsFilter />
  </div>
</div>
`;
